tests/posix_netdb/main.c

```diff
@@ -0,0 +1,294 @@
+    memset(&hints, 0, sizeof(struct addrinfo));
+    hints.ai_family = AF_UNSPEC + AF_INET + AF_INET6;
+    TEST_ASSERT_EQUAL_INT(EAI_FAMILY,
+                          getaddrinfo(SOCK_DNS_MOCK_EXAMPLE_COM_HOSTNAME, NULL, &hints, &result));
+    freeaddrinfo(result);
+}
+
+static void test_posix_netdb_getaddrinfo__EAI_NONAME(void)
+{
+    struct addrinfo *result = NULL;
+    struct addrinfo hints;
+
+    /* neither nodename nor servname given */
+    memset(&hints, 0, sizeof(struct addrinfo));
+    TEST_ASSERT_EQUAL_INT(EAI_NONAME, getaddrinfo(NULL, NULL, &hints, &result));
+    freeaddrinfo(result);
+
+    /* nodename and servname are empty strings */
+    memset(&hints, 0, sizeof(struct addrinfo));
+    TEST_ASSERT_EQUAL_INT(EAI_NONAME, getaddrinfo("", "", &hints, &result));
+    freeaddrinfo(result);
+
+    /* invalid IPv4 address */
+    memset(&hints, 0, sizeof(struct addrinfo));
+    hints.ai_flags |= AI_NUMERICHOST;
+    hints.ai_family = AF_INET;
+    TEST_ASSERT_EQUAL_INT(EAI_NONAME, getaddrinfo("1.2.3.4.5",NULL, &hints, &result));
+    freeaddrinfo(result);
+
+    /* invalid IPv6 address */
+    memset(&hints, 0, sizeof(struct addrinfo));
+    hints.ai_flags |= AI_NUMERICHOST;
+    hints.ai_family = AF_INET6;
+    TEST_ASSERT_EQUAL_INT(EAI_NONAME, getaddrinfo("1:2:3:4:5:6:7:8:9",NULL, &hints, &result));
+    freeaddrinfo(result);
+}
+
+static void test_posix_netdb_getaddrinfo__EAI_SERVICE(void)
+{
+    struct addrinfo *result = NULL;
+    struct addrinfo hints;
+
+    /* non-numeric port */
+    memset(&hints, 0, sizeof(struct addrinfo));
+    hints.ai_flags |= AI_NUMERICSERV;
+    TEST_ASSERT_EQUAL_INT(EAI_SERVICE, getaddrinfo(NULL, "abc", &hints, &result));
+    freeaddrinfo(result);
+
+    /* invalid port ( port < 0 ) */
+    memset(&hints, 0, sizeof(struct addrinfo));
+    hints.ai_flags |= AI_NUMERICSERV;
+    TEST_ASSERT_EQUAL_INT(EAI_SERVICE, getaddrinfo(NULL, "-1", &hints, &result));
+    freeaddrinfo(result);
+
+    /* invalid port ( port > 65535 ) */
+    memset(&hints, 0, sizeof(struct addrinfo));
+    hints.ai_flags |= AI_NUMERICSERV;
+    TEST_ASSERT_EQUAL_INT(EAI_SERVICE, getaddrinfo(NULL, "70000", &hints, &result));
+    freeaddrinfo(result);
+}
+
+static void test_posix_netdb_getaddrinfo__AI_NUMERICHOST(void)
+{
+    struct addrinfo *result = NULL;
+    struct addrinfo hints;
+
+    char* test_ipv4 = "1.2.3.4";
+    char addr4_str[INET_ADDRSTRLEN];
+
+    /* explicit IPv4 address */
+    memset(&hints, 0, sizeof(struct addrinfo));
+    hints.ai_flags |= AI_NUMERICHOST;
+    hints.ai_family = AF_INET;
+    TEST_ASSERT_EQUAL_INT(0, getaddrinfo(test_ipv4, NULL, &hints, &result));
+    TEST_ASSERT_NOT_NULL(result);
+    TEST_ASSERT_EQUAL_INT(AF_INET, result->ai_family);
+    TEST_ASSERT_NOT_NULL(inet_ntop(AF_INET, &((struct sockaddr_in*)(result->ai_addr))->sin_addr,
+                                   addr4_str, sizeof(addr4_str)));
+    TEST_ASSERT_EQUAL_INT(0, strcmp(addr4_str, test_ipv4));
```

Comment:
Here and in other instances where you used `TEST_ASSERT_EQUAL_INT(0, strcmp(...))`

```suggestion
    TEST_ASSERT_EQUAL_STRING(addr4_str, test_ipv4);
```